Source/ACE/Entity/Landblock.cs

```diff
@@ -819,19 +821,22 @@ private void ItemTransferContainerInternal(ObjectGuid woGuid, Container containe
 
             RemoveWorldObjectInternal(woGuid, false);
             wo.ContainerId = container.Guid.Full;
-            container.AddToInventory(wo);
+            // We are coming off the world we need to be ready to save.
+            wo.Location = null;
+            wo.InitializeAceObjectForSave();
+            container.AddToInventory(wo, placement);
         }
 
-        private void ItemTransferInternal(ObjectGuid woGuid, ObjectGuid containerGuid)
+        private void ItemTransferInternal(ObjectGuid woGuid, ObjectGuid containerGuid, uint placement = 0)
         {
             Container container = GetObject(containerGuid) as Container;
 
-            ItemTransferContainerInternal(woGuid, container);
+            ItemTransferContainerInternal(woGuid, container, placement);
         }
 
         private void Log(string message)
         {
-            log.Debug($"LB {id.Landblock.ToString("X")}: {message}");
+            log.Debug($"LB {id.Landblock:X}: {message}");
```

Comment:
The ToString was right - it would have printed the landblock Id in Hex.  You're now only logging the "X" component.